src/aurelia-slickgrid/filters/autoCompleteFilter.ts

```diff
@@ -352,67 +385,103 @@ export class AutoCompleteFilter implements Filter {
+      this._autoCompleteOptions = { ...definedOptions, ...this.filterOptions };
+      this.$filterElm.autocomplete(this._autoCompleteOptions);
+
+      // we'll use our own renderer so that it works with label prefix/suffix and also with html rendering when enabled
+      this.$filterElm.autocomplete('instance')._renderItem = this.renderCollectionItem.bind(this);
     }
 
-    $filterElm.val(searchTermInput);
-    $filterElm.data('columnId', columnId);
+    this.$filterElm.val(searchTermInput);
+    this.$filterElm.data('columnId', columnId);
 
     // if there's a search term, we will add the "filled" class for styling purposes
     if (searchTerm) {
-      $filterElm.addClass('filled');
+      this.$filterElm.addClass('filled');
     }
 
     // append the new DOM element to the header row
-    if ($filterElm && typeof $filterElm.appendTo === 'function') {
+    if (this.$filterElm && typeof this.$filterElm.appendTo === 'function') {
       const $container = $(`<div class="autocomplete-container"></div>`);
       $container.appendTo($headerElm);
-      $filterElm.appendTo($container);
+      this.$filterElm.appendTo($container);
 
       // add a <span> in order to add spinner styling
       $(`<span></span>`).appendTo($container);
     }
 
-    return $filterElm;
+    // we could optionally trigger a search when clicking on the AutoComplete
+    if (this.filterOptions.openSearchListOnFocus) {
+      this.$filterElm.click(() => this.$filterElm.autocomplete('search', this.$filterElm.val()));
+    }
+
+    // user might override any of the jQueryUI callback methods
+    if (this.columnFilter.callbacks) {
+      for (const callback of Object.keys(this.columnFilter.callbacks)) {
+        if (typeof this.columnFilter.callbacks[callback] === 'function') {
+          this.$filterElm.autocomplete('instance')[callback] = this.columnFilter.callbacks[callback];
+        }
+      }
+    }
+
+    return this.$filterElm;
   }
 
+
   //
   // private functions
   // ------------------
 
   // this function should be PRIVATE but for unit tests purposes we'll make it public until a better solution is found
   // a better solution would be to get the autocomplete DOM element to work with selection but I couldn't find how to do that in Jest
-  onSelect(event: Event, ui: any): boolean {
+  onSelect(event: Event, ui: { item: any; }) {
     if (ui && ui.item) {
-      const itemLabel = typeof ui.item === 'string' ? ui.item : ui.item.label;
-      const itemValue = typeof ui.item === 'string' ? ui.item : ui.item.value;
+      const item = ui.item;
+      // when the user defines a "renderItem" (or "_renderItem") template, then we assume the user defines his own custom structure of label/value pair
+      // otherwise we know that jQueryUI always require a label/value pair, we can pull them directly
+      const hasCustomRenderItemCallback = this.columnFilter?.callbacks?.hasOwnProperty('_renderItem') ?? this.columnFilter?.filterOptions?.renderItem ?? false;
```

Comment:
need to remove `?.`
`const hasCustomRenderItemCallback = this.columnFilter && this.columnFilter.callbacks && this.columnFilter.callbacks.hasOwnProperty('_renderItem') || (this.columnFilter && this.columnFilter.filterOptions && this.columnFilter.filterOptions.renderItem) || false;`